test/analytics/tasks/test_utils.py

```diff
@@ -0,0 +1,52 @@
+from zoo.analytics.tasks.utils import DockerImageId
+
+
+def test_docker_image_id_parsing_short():
+    parsed = DockerImageId("python:3.7")
+    assert parsed.full_image_id == "python:3.7"
+    assert parsed.registry is None
+    assert parsed.registry_host_name is None
+    assert parsed.registry_port is None
+    assert parsed.full_image_name == "python:3.7"
+    assert parsed.full_image_name_no_tag == "python"
+    assert parsed.username is None
+    assert parsed.full_repository_name == "python"
+    assert parsed.repository_name == "python"
+    assert parsed.namespaces is None
+    assert parsed.tag == "3.7"
+    assert parsed.version == "3.7"
+    assert parsed.full_os is None
+    assert parsed.os is None
+    assert parsed.os_version is None
+    assert parsed.alias is None
+
+
+def test_docker_image_id_parsing_long():
```

Comment:
Maybe some more test cases? With parts missing and so on

Reply:
Makes sense, I will add some random variations based on real examples

Reply:
Awesome, thank you very much 🙂

Reply:
Glad to hear that :)